Job board

catzkorn opened the pull request:
First itteration of a Job board. 

CRUD designed with additional methods that allow only approved job posts to be shown, and a method that can set the approved bool to true or false based on input.

Commit:
job_board schema

Fields as detailed in issue #15

Commit:
Create and Get Job methods

Add test to check functions behave as expected

Commit:
Update Job method

Future commit will include approved flag

Commit:
Delete Job method

Tests check whether a known job can be deleted, and whether a job that does not exist can be deleted

Commit:
ListJobs method

Method returns all jobs, approved or not

Commit:
ListApprovedJobs method

Method returns only approved jobs to users

Commit:
UpdateApproveJob method

sets the approved flag to true or false depending on input for a specific job id

bketelsen commented on conferences/deletejob_test.go:
```diff
@@ -0,0 +1,44 @@
+	"testing"
+)
+
+func TestDeletePaper(t *testing.T) {
```
change to TestDeleteJob

bketelsen commented on conferences/listjobs.go:
```diff
@@ -0,0 +1,69 @@
+			discord,
+			rank,
+			approved
+		FROM job_board
```
let's add order by rank here

bketelsen reviewed (changes requested):
This looks great, pending a few tiny changes

Commit:
Amended code for suggested changes

Removed copypasta fails of paper, now changed correctly to job

ListJobs and ListApprovedJobs now order by rank
